### PR TITLE
Dave/solo mode refactor

### PR DESCRIPTION
**Pull request by daveschmidt86**

### CODE REVIEW

**Comment by daveschmidt86** [comment]
Hey Jeff,

DId a bit of a refactor, I versioned the handlers and wrote some additional tests. The feature flag now have a app level config as well that allows for us to test configuration logic independently as well.

**Comment by gubbin** [comment]
Your review environment is available at http://review-hello-cammis-pr32.dev.cammis-test.com

**Comment by jeffmaher** [comment]
Accidentally did the tests with commits from yesterday (pre-v2). Will try again.

**Comment by jeffmaher** [comment]
With the freshly pulled code, I get this error on startup 

```
jeff@providence (~/ca-mmis/hello-cammis)
>docker run -it --rm -v "$(pwd)":"/hello-cammis" cadhcscammis/hello-cammis npm install && docker run -it --rm -p 3011:3011 -v "$(pwd)":"/hello-cammis" -e HELLO_CAMMIS_DATA_URL="http://host.docker.internal:8000" -e PORT=3011 cadhcscammis/hello-cammis
audited 527 packages in 5.981s
found 0 vulnerabilities


> hello-cammis@1.0.0 start /hello-cammis
> node index.js

/hello-cammis/lib/routes/index.js:6
const v1router = require('./v2');
      ^

SyntaxError: Identifier 'v1router' has already been declared
    at new Script (vm.js:79:7)
    at createScript (vm.js:251:10)
    at Object.runInThisContext (vm.js:303:10)
    at Module._compile (internal/modules/cjs/loader.js:656:28)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:699:10)
    at Module.load (internal/modules/cjs/loader.js:598:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:537:12)
    at Function.Module._load (internal/modules/cjs/loader.js:529:3)
    at Module.require (internal/modules/cjs/loader.js:636:17)
    at require (internal/modules/cjs/helpers.js:20:18)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! hello-cammis@1.0.0 start: `node index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the hello-cammis@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-02-11T17_13_17_108Z-debug.log
```

Steps Performed:

1. `sh docker/build-local.sh`
1. Get hello-cammis-data running on port 8000
1. `docker run -it --rm -v "$(pwd)":"/hello-cammis" cadhcscammis/hello-cammis npm install && docker run -it --rm -p 3011:3011 -v "$(pwd)":"/hello-cammis" -e HELLO_CAMMIS_DATA_URL="http://host.docker.internal:8000" -e PORT=3011 cadhcscammis/hello-cammis`

File: lib/routes/v1/healthcheck/healthcheck_handler.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+const healthcheck = require('./healthcheck');
+
+router.get('/', (req, res, next) => {
```
[the remark this answers is not in the document]
Comment (daveschmidt86) [reply lib/routes/v1/healthcheck/healthcheck_handler.js]: due to versioning, new routing coming soon

File: lib/routes/v1/hello/hello.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const axios = require('axios');
+
+const hello_solo = (name, callback, errorCallback) => {
+  callback(`Hello ${name}!`);
+};
+
+const hello_api = (name, config, callback, errorCallback) => {
```
[the remark this answers is not in the document]
Comment (daveschmidt86) [reply lib/routes/v1/hello/hello.js]: due to versioning, new routing coming soon

File: lib/routes/v2/hello/hello.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const axios = require('axios');
+
+const hello_solo = (name, callback, errorCallback) => {
+  callback(`Hello, welcome ${name}!`);
+};
+
+const hello_api = (name, config, callback, errorCallback) => {
```
[the remark this answers is not in the document]
Comment (daveschmidt86) [reply lib/routes/v2/hello/hello.js]: due to versioning, new routing coming soon

File: lib/routes/v1/hello/hello_handler.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+const hello = require('./hello');
+
+router.get('/:name', (req,res) => {
```
[the remark this answers is not in the document]
Comment (daveschmidt86) [reply lib/routes/v1/hello/hello_handler.js]: due to versioning, new routing coming soon

File: lib/routes/v2/hello/hello_handler.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+const hello = require('./hello');
+
+router.get('/:name', (req,res) => {
```
[the remark this answers is not in the document]
Comment (daveschmidt86) [reply lib/routes/v2/hello/hello_handler.js]: due to versioning, new routing coming soon

**Comment by daveschmidt86** [comment]
@jeffmaher updates made, please review again when you have a moment. Curious for any additional feedback.

File: README.md [hunk 6]
```diff
@@ -17,10 +17,17 @@ These all assume that hello-cammis-data is running on `localhost` at port `8000`
...
+
 ## Running Docker Image
 
 `docker run -p 3000:3000 -e HELLO_CAMMIS_DATA_HOST="<url>" -e HELLO_CAMMIS_DATA_PORT="<port>" cadhcscammis/hello-cammis`
 
 ### Optional environmental flags:
- 
+
 - `-e PORT=<port>`: Desired port number, will need to correspond to `-p` above
+- `-e HELLO_CAMMIS_DATA_URL=<hello-cammis-data-url>: URL to hello-cammis-data api (if not provided hello-cammis serves static local data "solo mode")`
```
Comment (jeffmaher) [inline README.md]:
```suggestion
- `-e HELLO_CAMMIS_DATA_HOST=<hello-cammis-data-hostname>: Hostname for hello-cammis-data (if not provided hello-cammis serves static local data "solo mode")`
- `-e HELLO_CAMMIS_DATA_PORT=<hello-cammis-data-port>: Port number for hello-cammis-data
```

File: lib/routes/route_map.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+      }
+    ]
+  },
+  v2: {
+    routes: [
+      {
+        method: "GET",
+        endpoint: "/hello/:name",
+        handler: "v2/hello"
+      }
```
Comment (jeffmaher) [inline lib/routes/route_map.js]: Just verifying: v2 doesn't have a health check yet?

Comment (daveschmidt86) [reply lib/routes/route_map.js]: route_map.js was not supposed to be included. I need to pull that file out, currently not using this.

**Pull request merged**